modules/data-entry/src/main/frontend/src/questionnaire/FormPageNavigation.jsx

```diff
@@ -0,0 +1,158 @@
+//  under the License.
+//
+
+import React, { useState, useEffect } from "react";
+
+import {
+  Fab,
+  Grid,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+  Tooltip,
+} from "@mui/material";
+
+import CheckIcon from '@mui/icons-material/Check';
+import WarningIcon from '@mui/icons-material/Warning';
+
+import { useTheme } from '@mui/material/styles';
+import useMediaQuery from '@mui/material/useMediaQuery';
+
+import PropTypes from "prop-types";
+
+import FormattedText from "../components/FormattedText";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+/**
+ * Component that enables the navigation between different pages of a Form. Used in FormPagination when the paginationVariant is "navigable".
+ */
+function FormPageNavigation (props) {
+  const { pages, activePage, saveButton, backButton, isPageCompleted, navigateTo } = props;
+
+  const [pageSelectorAnchorEl, setPageSelectorAnchorEl] = useState();
+
+  const theme = useTheme();
+  const condensedPageList = useMediaQuery(theme.breakpoints.down('lg'));
+
+  // -------------------------------------------------------------------------------
+  // Rendering helpers
+  //
+
+  // Page icon
+  // Previous pages that are enabled are marked with a checkmark if completed, and with a warning sign if incomplete
+  let pageIcon = (page, index) => (
+    index < activePage && page.canBeVisible && isPageCompleted
+    ? (isPageCompleted(page.keys) ? <CheckIcon sx={{fontSize: "medium"}} /> : <WarningIcon sx={{fontSize: "medium"}} />)
+    : (index + 1)
+  )
+
+  // Format the title
+  let pageTitle = (page) => (page ? <FormattedText variant="caption">{page.title}</FormattedText> : page)
+
+  // Individual page buttons
+  let pageButton = (page, index) => (
+    <Fab
+      size="small"
+      type="submit"
+      sx={{ width: 30, height: 30, minHeight: 30, boxShadow: "none", fontWeight: "normal" }}
+      color={index == activePage ? "primary" : ''}
+      onClick={() => navigateTo(index)}
+      disabled={!page.canBeVisible}
+    >
+      { pageIcon(page, index) }
+    </Fab>
+  )
+
+  // List of page buttons
+  let pageList = () => (
+    <Grid container spacing={2} justifyContent="space-evenly">
+    { pages.map((p, index) => (
+      <Grid item key={index}>
+      { p.canBeVisible ?
+          <Tooltip title={pageTitle(p)}>
+          { pageButton(p, index) }
+          </Tooltip>
+        :
+          pageButton(p, index)
+      }
+      </Grid>
+    ))}
+    </Grid>
+  );
+
+  // Condensed view: one button that unravels a page "menu"
+  let pageSelector = () => (
+    <div>
+      <Fab
+        sx={{ boxShadow: "none", fontWeight: "normal" }}
+        onClick={event => setPageSelectorAnchorEl(event.currentTarget)}
+      >
+        { (activePage + 1) + "/" + pages.length }
+      </Fab>
+      <Menu
+        disablePortal
+        anchorEl={pageSelectorAnchorEl}
+        open={!!pageSelectorAnchorEl}
+        onClose={() => setPageSelectorAnchorEl(null)}
```

Comment:
```suggestion
        onClose={() => setPageSelectorAnchorEl(null)}
        disableScrollLock={true}
```
Having scrolling disabled when the menu is open feels awkward to me

Reply:
Done in 426482a .